atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/pool/CassandraService.java

```diff
@@ -437,8 +437,13 @@ public InetSocketAddress getRandomHostForKey(byte[] key) {
                         + " Falling back to choosing a random live node."
                         + " Current host blacklist is {}."
                         + " Current state logged at TRACE",
-                SafeArg.of("blacklistedHosts", blacklist.blacklistDetails()));
-        log.trace("Current ring view is: {}.", SafeArg.of("tokenMap", getRingViewDescription()));
+                SafeArg.of("blacklistedHostsCount", blacklist.size()));
+
+        // The following trace can cause high memory pressure
+        log.trace("Current ring view is: {}."
+                        + " Current host blacklist is {}.",
+                SafeArg.of("tokenMap", getRingViewDescription()),
+                SafeArg.of("blacklistedHostsCount", blacklist.size()));
```

Comment:
since we're concerned about allocation/GC overhead, we should only do this work when trace is enabled

```suggestion
        if (log.isTraceEnabled()) {
            log.trace("Current ring view is: {}."
                            + " Current host blacklist is {}.",
                    SafeArg.of("tokenMap", getRingViewDescription()),
                    SafeArg.of("blacklistedHostsCount", blacklist.size()));
        }
```

Reply:
yep absolutely